### PR TITLE
Use explicit version tag in Dockerfile

### LINKED ISSUE #851: Dockerfile should not use FROM microsoft/dotnet:latest
### Short issue description

The generated Dockerfile uses the `FROM microsoft/dotnet:latest` to indicate the source image. If there is a `microsoft/dotnet:latest` Docker image available it will use that one even if it is not the correct version.

On my laptop the `microsoft/dotnet:latest` tag pointed to a Docker image with Core CLR 1.0.1. This results in an error when doing a `docker build` where the build stops with `The command 'dotnet restore' returned a non-zero code: 139`

### Expected behavior

The Dockerfile to specify the correct version of the base image with `FROM microsoft/dotnet:1.1.0-sdk-projectjson`. This results in a correct build and image.

### Actual behavior
Docker builds using whatever image is tagged `microsoft/dotnet:latest` and fails if this is not correct.

### Steps to reproduce the behavior

1. Make sure you have a docker image with CLR 1.0.1 tagged as latest. Im my case this was image with ID 773ea153b89f
1. Create a new Web Application with `yo aspnet`. I used the Web Application Basic [without Membership and Authorization] but the same appears to be true for other options.
1. Run `docker build -t oops .`
1. Observe the output:
`
Step 1/7 : FROM microsoft/dotnet:latest
 ---> 773ea153b89f
Step 2/7 : COPY . /app
 ---> Using cache
 ---> ded8cf51581e
Step 3/7 : WORKDIR /app
 ---> Using cache
 ---> 6f4a26c24ce3
Step 4/7 : RUN dotnet restore
 ---> Running in 4db0a5b620d8
log  : Restoring packages for /app/project.json...
The command 'dotnet restore' returned a non-zero code: 139
`

----
#### OS version (Win/Mac/\*Nix?)
Windows 10

#### NodeJS version
v6.9.0

#### Yeoman version
1.8.5

#### Generator version
generator-aspnet@0.2.5

### PR DESCRIPTION
**Pull request by mauricedb**
Fixes #851.

Summary of the changes in this PR:
 - Use an explicit CLR version tag in the Dockerfile
 - Make sure the CLR version used in the project.json and Dockerfile match

Thanks!

/cc
@OmniSharp/generator-aspnet-team-push

### CODE REVIEW

File: templates/Dockerfile.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM microsoft/dotnet:latest
+FROM microsoft/dotnet:dotnet:1.1.0-sdk-projectjson
```
Comment (peterblazejewicz) [inline templates/Dockerfile.txt]: There is a typo I believe here. Should be:
```Dockerfile
FROM microsoft/dotnet:1.0-sdk-projectjson
```
😄

File: test/subgenerators.js [hunk 2]
```diff
@@ -90,6 +90,18 @@ describe('Subgenerators without arguments tests', function() {
...
+  describe('aspnet:dockerfile has the same .NET version', function() {
+    var sdkVersion = '1.1.0';
+    var arg = 'file';
+    var dir = util.makeTempDir();
+
+    util.goCreateApplication('web', 'webTest', dir);
+    util.goCreateWithArgs('mvccontroller', [arg], path.join(dir, 'webTest'));
+
+    util.fileContentCheck('project.json', 'file content check', new RegExp('"Microsoft.NETCore.App":\\s*{\\s*"version": "' + sdkVersion + '"'));
+    util.fileContentCheck('Dockerfile', 'Check the content for dotnet latest image tag', new RegExp('FROM microsoft\/dotnet:dotnet:' + sdkVersion + '-sdk-projectjson'));
```
Comment (peterblazejewicz) [inline test/subgenerators.js]: again, should be:
```JavaScript
'FROM microsoft\/dotnet:'
```

File: test/subgenerators.js [hunk 3]
```diff
@@ -99,7 +111,7 @@ describe('Subgenerators without arguments tests', function() {
     var filename = 'Dockerfile';
     util.goCreate(filename.toLowerCase());
     util.fileCheck('should create Dockerfile', filename);
-    util.fileContentCheck(filename, 'Check the content for dotnet latest image tag', /FROM microsoft\/dotnet:latest/);
+    util.fileContentCheck(filename, 'Check the content for dotnet latest image tag', /FROM microsoft\/dotnet:dotnet:/);
```
Comment (peterblazejewicz) [inline test/subgenerators.js]: As above `'FROM microsoft\/dotnet:'`

File: test/subgenerators.js [hunk 4]
```diff
@@ -109,7 +121,7 @@ describe('Subgenerators without arguments tests', function() {
     var filename = 'Dockerfile';
     util.goCreateWithArgs(filename.toLowerCase(), [arg]);
     util.fileCheck('should create Dockerfile', filename);
-    util.fileContentCheck(filename, 'Check the content for dotnet latest image tag', /FROM microsoft\/dotnet:latest/);
+    util.fileContentCheck(filename, 'Check the content for dotnet latest image tag', /FROM microsoft\/dotnet:dotnet:/);
```
Comment (peterblazejewicz) [inline test/subgenerators.js]: and last one `'FROM microsoft\/dotnet:'`

**Comment by mauricedb** [comment]
@peterblazejewicz

Oops copy and fail error with the dotnet:dotnet: :-(

I also added the Windows Nano Server dockerfile under `Dockerfile.nano` 

However I am unsure about the [dockerfile sub generator](https://github.com/OmniSharp/generator-aspnet/blob/master/README.md#dockerfile). I could add it there so it would add two files, one for Linux and one for Windows Nano. Else I could create a new sub generator `dockerfilenano`. But the `--sqlite` feels bit weird as it has nothing to do with SQLite but runs an EF database update.

**Comment by peterblazejewicz** [comment]
> However I am unsure about the dockerfile sub generator. I could add it there so it would add two files, one for Linux and one for Windows Nano. Else I could create a new sub generator dockerfilenano. But the --sqlite feels bit weird as it has nothing to do with SQLite but runs an EF database update.

Let's left it as is - as Docker support and usage becomes more mature it will be very likely done here by someone or by other repo with templates. The most important are correct images at the moment and they work. Here is screen from build and run image:

image

Thanks!

**Pull request merged**